server/src/test/java/org/elasticsearch/cluster/routing/allocation/AllocationCommandsTests.java

```diff
@@ -170,31 +173,59 @@ public void testAllocateCommand() {
 
         logger.info("--> allocating to non-existent node, should fail");
         try {
-            allocation.reroute(clusterState, new AllocationCommands(randomAllocateCommand(index, shardId.id(), "node42")), false, false);
+            allocation.reroute(
```

Comment:
I wonder if (at least in this test suite) we should be using the desired balance allocator rather than just `BalancedShardsAllocator`?

Reply:
https://github.com/elastic/elasticsearch/pull/89428

I have tried the code from above pr here and this test has passed

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/88848/commits/35a5b0e9e9a2477d4a0711715cfd13d4d8f2ffb6